### PR TITLE
Launch notebooks inside Saturn

### PR DESCRIPTION
**Pull request by MattPutnam**
Lots of whitespace-only changes.

### CODE REVIEW

**Review by panentheos — APPROVED** [review]
Looks good.

**Comment by zarsky-broad** [comment]
I notice that when it starts opening the notebook, if the cluster is stopped, it doesn't change visible status to starting until its normal poll. This makes it seem like it's not starting; can we jumpstart that?

**Comment by zarsky-broad** [comment]
Also, we shouldn't release this until we include a solution for needing to force clusters to regenerate in order to be embeddable.

**Comment by MattPutnam** [comment]
I wonder if Redux would be the right way to do that. They're two disparate components that both care about and modify the same state. @panentheos it shouldn't be ugly to use Redux just in a few isolated spots like this, right?

**Comment by zarsky-broad** [comment]
Couldn't we just use `Utils.connectAtom` and have everything delegate updating out to a common function call?

**Comment by panentheos** [comment]
Redux would work, yes, but I wouldn't add it just for this. You could also move the state to an atom to accomplish the same thing for this single piece of data.

**Comment by MattPutnam** [comment]
This is way more complex than I thought and is blowing up the scope. Let's do this as a separate ticket.

**Review by zarsky-broad — APPROVED** [review]

File: src/libs/auth.js [hunk 1]
```diff
@@ -64,9 +65,20 @@ authStore.subscribe(async (state, oldState) => {
...
 
+      const needsUpgrade = _.remove(c => c.labels.saturnVersion === version, clusters)
+      await Promise.all(_.map(({ googleProject, clusterName, machineConfig, jupyterUserScriptUri }) => {
+        const cluster = Leo.cluster(googleProject, clusterName)
+        const clusterOptions = {
+          labels: { 'saturnAutoCreated': 'true', 'saturnVersion': version },
+          machineConfig,
+          jupyterUserScriptUri
+        }
+        return cluster.delete().then(() => cluster.create(clusterOptions))
```
Comment (panentheos) [inline src/libs/auth.js]: This will re-use the same name. Is that safe, or should we generate a new one?

Comment (MattPutnam) [reply src/libs/auth.js]: I'm not sure, but I think it's safe.

Comment (MattPutnam) [reply src/libs/auth.js]: I'm not finding any problems.

Comment (zarsky-broad) [reply src/libs/auth.js]: I _think_ you should be just fine.

Comment (zarsky-broad) [reply src/libs/auth.js]: Alternately, if we create one with a new name, we wouldn't have to wait for the delete to finish to start creating it, which might be better.

File: src/pages/workspaces/workspace/notebooks/NotebookLauncher.js [hunk 2]
```diff
@@ -70,7 +78,7 @@ class NotebookLauncherContent extends Component {
   }
 
   async startCluster() {
-    while (true) {
+    while (this.mounted) {
```
Comment (zarsky-broad) [inline src/pages/workspaces/workspace/notebooks/NotebookLauncher.js:81]: Could we just do `while(this)`?

Comment (MattPutnam) [reply src/pages/workspaces/workspace/notebooks/NotebookLauncher.js:81]: No, the whole object continues to exist

**Pull request merged**